goldmane/proto/api.proto

```diff
@@ -1,36 +1,29 @@
 syntax = "proto3";
 
-package felix;
+package goldmane;
 
 option go_package = "./proto";
 
-service FlowAPI {
+// FlowService provides APIs for querying aggregated Flow data.
+//
+// The returned Flows will be aggregated across cluster nodes, as well as the specified aggregation
+// time interval.
+service FlowService {
   // List is an API call to query for one or more Flows.
-  // Matching Flows are streamed back to the caller.
-  rpc List(FlowRequest) returns (stream Flow);
+  rpc List(ListRequest) returns (stream FlowResult);
 
-  // Stream is an API call to return a long running stream of new Flows as
-  // they are generated.
-  rpc Stream(FlowRequest) returns (stream Flow);
-}
+  // Stream is an API call to return a long running stream of new Flows as they are generated.
+  rpc Stream(StreamRequest) returns (stream FlowResult);
 
-// FlowCollector represents an API capable of receiving streams of Flow data
-// from cluster nodes.
-service FlowCollector {
-  // Connect receives a connection that may stream one or more FlowUpdates. A FlowReceipt is returned
-  // to the client by the server after each FlowUpdate.
-  //
-  // Following a connection or reconnection to the server, clients should duplicates of previously transmitted FlowsUpdates
-  // in order to allow the server to rebuild its cache, as well as any new FlowUpdates that have not previously been transmitted.
-  // The server is responsible for deduplicating where needed.
-  rpc Connect(stream FlowUpdate) returns (stream FlowReceipt);
+  // FilterHints can be used to discover available filter criteria, such as
+  // Namespaces and source / destination names. It allows progressive filtering of criteria based on
+  // other filters. i.e., return the flow destinations given a source namespace.
+  // Note that this API provides hints to the UI based on past flows and other values may be valid.
+  rpc FilterHints(FilterHintsRequest) returns (stream FilterHint);
 }
 
-// FlowReceipt is a response from the server to a client after publishing a Flow.
-message FlowReceipt {}
-
-// FlowRequest defines a message to request a particular selection of aggregated Flow objects.
-message FlowRequest {
+// ListRequest defines a message to request a particular selection of aggregated Flow objects.
+message ListRequest {
```

Comment:
From a general services design POV, it might be worth it to prefix this with `Flow`. There could be other services with listing abilities.